ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/block/ReexecutingExecutionPayloadBlockManager.java

```diff
@@ -0,0 +1,172 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.statetransition.block;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+import tech.pegasys.teku.infrastructure.async.Cancellable;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.collections.LimitedSet;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.spec.logic.common.statetransition.results.BlockImportResult;
+import tech.pegasys.teku.spec.logic.common.statetransition.results.BlockImportResult.FailureReason;
+import tech.pegasys.teku.statetransition.util.FutureItems;
+import tech.pegasys.teku.statetransition.util.PendingPool;
+import tech.pegasys.teku.statetransition.validation.BlockValidator;
+import tech.pegasys.teku.storage.client.RecentChainData;
+
+public class ReexecutingExecutionPayloadBlockManager extends BlockManager {
+  private static final Logger LOG = LogManager.getLogger();
+  private static final int EXPIRES_IN_SLOT = 2;
+  private static final Duration RECHECK_INTERVAL = Duration.ofSeconds(2);
+
+  private final AtomicBoolean reexecutingExecutionPayload = new AtomicBoolean(false);
+  private final Set<SignedBeaconBlock> pendingBlocks = LimitedSet.create(10);
+  private final AsyncRunner asyncRunner;
+
+  private Optional<Cancellable> cancellable = Optional.empty();
+
+  private ReexecutingExecutionPayloadBlockManager(
+      RecentChainData recentChainData,
+      BlockImporter blockImporter,
+      PendingPool<SignedBeaconBlock> pendingBlocks,
+      FutureItems<SignedBeaconBlock> futureBlocks,
+      BlockValidator validator,
+      final AsyncRunner asyncRunner) {
+    super(recentChainData, blockImporter, pendingBlocks, futureBlocks, validator);
+    this.asyncRunner = asyncRunner;
+  }
+
+  public static BlockManager create(
+      final PendingPool<SignedBeaconBlock> pendingBlocks,
+      final FutureItems<SignedBeaconBlock> futureBlocks,
+      final RecentChainData recentChainData,
+      final BlockImporter blockImporter,
+      final BlockValidator validator,
+      final AsyncRunner asyncRunner) {
+    return new ReexecutingExecutionPayloadBlockManager(
+        recentChainData, blockImporter, pendingBlocks, futureBlocks, validator, asyncRunner);
+  }
```

Comment:
I think we can skip the static factory method since it's just executing the constructor directly (they're worth having if they do some kind of work/create things before calling the constructor but not otherwise). We do have a bunch like this that aren't necessary already in the code base but are moving away from them (feel free to delete pointless ones you come across).

Reply:
The conversation was marked as resolved but the factory method is still here. Might have missed pushing something?

Reply:
ops. removed usage but not the actual method.